### PR TITLE
[tests] fix "Tester object has no attribute '_testMethodName'"

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
I am not sure whether this is related to my test environment, but I got a lot of these kinds of failures when running tests as shown below: 

```bash
============================================================= ERRORS ==============================================================
__________________________ ERROR collecting tests/models/pix2struct/test_image_processing_pix2struct.py ___________________________
/usr/lib/python3.10/unittest/case.py:461: in __hash__
    return hash((type(self), self._testMethodName))
E   AttributeError: 'Pix2StructImageProcessingTester' object has no attribute '_testMethodName'
_____________________________ ERROR collecting tests/models/pixtral/test_image_processing_pixtral.py ______________________________
/usr/lib/python3.10/unittest/case.py:461: in __hash__
    return hash((type(self), self._testMethodName))
E   AttributeError: 'PixtralImageProcessingTester' object has no attribute '_testMethodName'
__________________________ ERROR collecting tests/models/pop2piano/test_feature_extraction_pop2piano.py ___________________________
/usr/lib/python3.10/unittest/case.py:461: in __hash__
    return hash((type(self), self._testMethodName))
E   AttributeError: 'Pop2PianoFeatureExtractionTester' object has no attribute '_testMethodName'
======================================================== warnings summary =========================================================
``` 
After the fix, tests run through. Could you take a look? Thanks a lot!

@amyeroberts @ydshieh

### CODE REVIEW

**Comment by Rocketknight1** [comment]
I'm not sure we've observed this on our own test runners, but cc @ydshieh anyway

**Comment by ydshieh** [comment]
Hi @faaany . Could you provide the command that you used to launch the test? Thank you

**Comment by ydshieh** [comment]
But IMO, those classes like `SegformerImageProcessingTester` should not be a subclass of `unittest.Testcase`. They are not the classes to be tested but just helpers for the corresponding classes like `SegformerImageProcessingTest`.

Don't know why they are like that 😅

**Comment by faaany** [comment]
> Hi @faaany . Could you provide the command that you used to launch the test? Thank you

Below is my command example. Actually nothing special, I used to run with this command before and it worked. Could you help run this command on your machine and see how it goes? 

```bash
python -m pytest tests/models/s*
```

**Comment by ydshieh** [comment]
Sure. But I guess you mean `python -m pytest tests/models/*`?
BTW, in our CI workflow, each model is run in a separate (GitHub Action) job, rather than launched in the same process (like you do here), so there might have some difference.

**Comment by faaany** [comment]
> Hmm, not able to reproduce it in our CI env. But as mentioned, IMO, the changes should be remove the `unittest.TestCase` part in `class ASTFeatureExtractionTester(unittest.TestCase):` and similar place.
> 
> @faaany Would you be up to do this change? After that, we can check again if there is still some issues (running in your env.) and see what extra would be needed to do.

Sure, good ideas! I will try this out.

**Comment by faaany** [comment]
@ydshieh yes, after removing `unittest.TestCase`, it works in my env

**Comment by faaany** [comment]
done. it works.

**Review by ydshieh — APPROVED** [review]
Glad it works in you env now, `XPU` must be happy this Friday!

Thanks again.

**Comment by ydshieh** [comment]
@Rocketknight1 Would you like to take final look and maybe give us a ✅ 🙏 ?

TL;DR: the `Tester` class is not the class to be tested, they are just some helpers.

File: tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py
Comment (ydshieh) [inline tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py]: I think we can remove `()`, right?

Comment (faaany) [reply tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py]: yes

Comment (faaany) [reply tests/models/audio_spectrogram_transformer/test_feature_extraction_audio_spectrogram_transformer.py]: I think it got removed in style fixing.

**Review by Rocketknight1 — APPROVED** [review]
This looks clean to me, but it's big enough that I should get a final review from a core maintainer! cc @ArthurZucker @LysandreJik 

Quick summary for them: Several "Tester" classes were subclasses of `unittest.TestCase`, but didn't actually have tests - they were just helpers for the actual test classes. This PR removes that subclassing, which fixes some issues.

**Comment by faaany** [comment]
I think the failed tests are unrelated to my changes. Could you help retrigger the CI, when it passes?

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks

**Pull request merged**
